tools/config-gen/params.yaml

```diff
@@ -15,16 +31,308 @@
+    How long the directory listing (output of ls <dir>) should be cached in the
+    kernel page cache. If a particular directory cache entry is kept by kernel
+    for longer than TTL, then it will be sent for invalidation by gcsfuse on
+    next opendir (comes in the start, as part of next listing) call. 0 means no
+    caching. Use -1 to cache for lifetime (no ttl). Negative value other than -1
+    will throw error.
+
+- flag-name: "http-client-timeout"
+  config-path: "gcs-connection.http-client-timeout"
+  type: "duration"
+  usage: >-
+    The time duration that http client will wait to get response from the
+    server. The default value 0 indicates no timeout.
+
+- flag-name: "max-retry-duration"
+  type: "duration"
+  usage: "This is currently unused."
+  deprecated: true
+  deprecation-warning: "This is currently unused."
+
+- flag-name: "retry-multiplier"
+  config-path: "gcs-retries.multiplier"
+  type: "float64"
+  usage: Param for exponential backoff algorithm, which is used to increase waiting time b/w two consecutive retries.
+  default: 2
+
+- flag-name: "temp-dir"
+  config-path: "file-system.temp-dir"
+  type: "string"
+  usage: >-
+    Path to the temporary directory where writes are staged prior to upload to
+    Cloud Storage. (default: system default, likely /tmp)"
+
+- flag-name: "client-protocol"
+  config-path: "gcs-connection.client-protocol"
+  type: "protocol"
+  usage: >-
+    The protocol used for communicating with the GCS backend.
+    Value can be 'http1' (HTTP/1.1), 'http2' (HTTP/2) or 'grpc'.
+  default: "http1"
+
+- flag-name: "max-conns-per-host"
+  config-path: "gcs-connection.max-conns-per-host"
+  type: "int"
+  usage: >-
+    The max number of TCP connections allowed per server. This is effective when
+    client-protocol is set to 'http1'. The default value 0 indicates no limit on
+    TCP connections (limited by the machine specifications).
+  default: "0"
+  
+- flag-name: "max-idle-conns-per-host"
+  config-path: "gcs-connection.max-idle-conns-per-host"
+  type: "int"
+  usage: "The number of maximum idle connections allowed per server."
+  default: "100"
+
+- flag-name: "enable-nonexistent-type-cache"
+  config-path: "metadata-cache.enable-nonexistent-type-cache"
+  type: "bool"
+  usage: >-
+    Once set, if an inode is not found in GCS, a type cache entry with type
+    NonexistentType will be created. This also means new file/dir created might
+    not be seen. For example, if this flag is set, and metadata-cache-ttl-secs
+    is set, then if we create the same file/node in the meantime using the same
+    mount, since we are not refreshing the cache, it will still return nil.
+
+- flag-name: "stackdriver-export-interval"
+  config-path: "metrics.stackdriver-export-interval"
+  type: "duration"
+  usage: >-
+    Export metrics to stackdriver with this interval. The default value 0
+    indicates no exporting.
+  default: "0s"
+
+- flag-name: "experimental-opentelemetry-collector-address"
```

Comment:
Shall we skip this from config file as it is deprecated?
same for other flags like `experimental-enable-json-read`

Reply:
We still need to have a config-path for it in order to fetch the value, right?